### DISCUSSION TITLE
Packaging into .exe?

### DISCUSSION THREAD

**Discussion by nixshal** [jac0b-w/album-art-wallpaper#15] [Q&A] [answered]
Lovely project and thank you for sharing!

Could I know how you packaged it into an executable? I find myself confused with the multiple sources available on this. Would appreciate a simple walkthrough or pointing me in the right direction! Thank you

**Comment by jac0-b** [comment, marked as answer]
Yeah so packaging a python script is usually relatively easy you can use several different programs to do this. I use [pyinstaller](https://www.pyinstaller.org/) and target my main script. This works fine for just python code but if you have assets such as icons/images then it doesn't package it all into one file (not by default at least). Also I have found a couple dependencies a pain to get working specifically scipy and sklearn so most of the time it works seamlessly. 

For this project I then create an installer using [inno setup](https://jrsoftware.org/isinfo.php) which has a bunch of features. It automatically adds your exe to the start menu + adds uninstaller and you can set it run your exe at startup which is very useful for AlbumPaper.

**Reply by nixshal** [reply]
Thank you @jac0b-w ! Appreciate it

**Reply by jac0-b** [reply]
No problem!
